server/src/main/java/org/elasticsearch/gateway/AsyncShardFetch.java

```diff
@@ -277,13 +290,9 @@ private List<NodeEntry<T>> findNodesToFetch(Map<String, NodeEntry<T>> shardCache
     /**
      * Are there any nodes that are fetching data?
      */
-    private boolean hasAnyNodeFetching(Map<String, NodeEntry<T>> shardCache) {
-        for (NodeEntry<T> nodeEntry : shardCache.values()) {
-            if (nodeEntry.isFetching()) {
-                return true;
-            }
-        }
-        return false;
+    // visible for testing
```

Comment:
I'm ok with keeping this private, I don't think it needs testing.

Reply:
++ modified